server/cli/init/base-file-sources/world.js

```diff
@@ -23,20 +24,46 @@ module.exports = function () {
     };
 
     /* eslint-disable new-cap */
-    this.Before(function (callback) {
+    this.Before(function (scenario, callback) {
     /* eslint-enable new-cap */
         global.httpBackend = new HttpBackend(global.browser);
         callback();
     });
 
     /* eslint-disable new-cap */
-    this.After(function (callback) {
+    this.After(function (scenario, callback) {
+        var browserlogArray = [];
+        var message;
     /* eslint-enable new-cap */
         global.httpBackend.clear();
-        global.browser.manage.deleteAllCookies();
+        global.browser.manage().deleteAllCookies();
         global.browser.executeScript('window.sessionStorage.clear();');
         global.browser.executeScript('window.localStorage.clear();');
-        callback();
+        if (scenario.isFailed()) {
+            global.browser.takeScreenshot().then(function (base64png) {

```

Comment:
Let's break all of this out into a different function. `createReports` or something.